File: Back-end/src/main/java/co/edu/uniandes/isis2503/nosqljpa/service/Reporte.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package co.edu.uniandes.isis2503.nosqljpa.service;
import co.edu.uniandes.isis2503.nosqljpa.logic.ReporteLogic;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

@Produces(MediaType.APPLICATION_JSON)
@Path("/reporte")
public class Reporte {
     public Reporte()
    {
        this.reporteLogic = new ReporteLogic();
    }
    
     private ReporteLogic reporteLogic; 
     
     
    @GET
    public String getReporte() {
        
        
        return reporteLogic.getReport();
    }
}

File: Front-with-auth0/04-Authorization/src/Funcionalidades/Reporte.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
import { Table, Button, FormGroup, ControlLabel, FormControl, Glyphicon, Pagination } from 'react-bootstrap';
import { Breadcrumb, BreadcrumbItem } from 'reactstrap';
import axios from 'axios';
import React from 'react';

class Reporte extends React.Component {
    
    
    getblah() {
        axios.get('http://localhost:8084/reporte')
                .then(response => this.setState({reporte: response.data.medidas}))
                .catch(error => this.setState({message: error.message}));
    }
  render() {
        return (
                <div>  
           <Breadcrumb>
        <BreadcrumbItem><a href="por">Funcionalidades</a></BreadcrumbItem>
        <BreadcrumbItem active>Reporte</BreadcrumbItem>
      </Breadcrumb>
        <Button color="primary">Reporte</Button>{' '}
        <table>
            <tr>
              <th>Reporte</th>
            </tr>
            <tr>
              <td>Numero de Alertas Fuera De Rango</td>
              <td>reporte.AFR</td>
            </tr>
            <tr>
              <td>Numero de Alertas Fuera de Linea</td>
              <td>reporte.AFL</td>
            </tr>
            <tr>
              <td>Promedio Ruido</td>
              <td>reporte.Ruido</td>
            </tr>
            <tr>
              <td>Promedio Temperatura</td>
              <td>reporte.Temperatura</td>
            </tr>
            <tr>
              <td>Promedio Iluminacion</td>
              <td>reporte.Luz</td>
            </tr>
            <tr>
              <td>Promedio Gases</td>
              <td>reporte.CO</td>
            </tr>
          </table>
             
         
                </div>


                );
    }
}
export default Reporte;



